api/src/test/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAOTest.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Assert;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.rules.ExpectedException;
+import org.openmrs.*;
+import org.openmrs.api.db.DiagnosisDAO;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateDiagnosisDAOTest extends BaseContextSensitiveTest {
+
+	private static final String DIAGNOSIS_XML = "org/openmrs/api/include/HibernateDiagnosisDAOTestDataset.xml";
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Autowired
+	DiagnosisDAO diagnosisDAO;
+
+	@Before
+	public void setUp() {
+		executeDataSet(DIAGNOSIS_XML);
+
+	}
+
+	@Test
+	public void shouldSaveDiagnosis() {
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText(new Concept(4),
+			new ConceptName(5089), "non coded");
+		Date date = new Date();
+		int id = 4;
+		String uuid = "4e663d96-6b78-11e0-93c3-18a9b5e044dc";
+		int rank = 2;
+		ConditionVerificationStatus certainty = ConditionVerificationStatus.CONFIRMED;
+		User creator = new User(1);
+		Condition condition = new Condition();
+		Patient patient = new Patient(2);
+		Encounter encounter = new Encounter(3);
+		
+		Diagnosis diagnosis = new Diagnosis();
+		diagnosis.setEncounter(encounter);
+		diagnosis.setRank(rank);
+		diagnosis.setId(id);
+		diagnosis.setCertainty(certainty);
+		diagnosis.setPatient(patient);
+		diagnosis.setCondition(condition);
+		diagnosis.setDiagnosis(codedOrFreeText);
+		diagnosis.setUuid(uuid);
+		diagnosis.setCreator(creator);
+		diagnosis.setVoided(false);
+		diagnosis.setDateCreated(date);
+
+		diagnosisDAO.saveDiagnosis(diagnosis);
+
+		Diagnosis savedDiagnosis = diagnosisDAO.getDiagnosisById(id);
+		
+		assertEquals(uuid, diagnosis.getUuid());
+		assertEquals(id, (int) diagnosis.getDiagnosisId());
```

Comment:
I think this check is brittle, this will always break whenever the test dataset is updated to add new rows to this table